src/components/home/elements/WithdrawToken.vue

```diff
@@ -0,0 +1,637 @@
+                api: this.coldApi(),
+                opc: OPC_FUNCTION,
+                address: this.address,
+                // senderPublicKey: this.coldAddresses[this.address].publicKey,
+                fee: this.fee * VSYS_PRECISION,
+                feeScale: FEE_SCALE,
+                timestamp: Date.now(),
+                attachment: '',
+                contractId: this.contractId,
+                functionId: this.isSplit ? WITHDRAW_FUNCIDX_SPLIT : WITHDRAW_FUNCIDX,
+                function: transaction.prepareWithdraw(this.contractId, this.address, BigNumber(this.amount).multipliedBy(this.tokenUnity)),
+                functionExplain: 'Withdraw ' + this.amount + ' token from ' + this.contractId
+            }
+        }
+    },
+    methods: {
+        inputAmount(num) {
+            return BigNumber(num)
+        },
+        coldApi: function() {
+            return API_VERSION
+        },
+        isSubmitDisabled() {
+            return !(!this.isInsufficient() && this.isAmountValid() && this.isValidContractId(this.contractId))
+        },
+        sendData: function(walletType) {
+            let apiSchema
+            if (walletType === 'hotWallet') {
+                if (this.hasConfirmed) {
+                    return
+                }
+                this.hasConfirmed = true
+                this.fee = BigNumber(CONTRACT_EXEC_FEE)
+                this.feeScale = FEE_SCALE
+                const dataInfo = {
+                    contractId: this.contractId,
+                    senderPublicKey: this.getKeypair(this.addresses[this.address]).publicKey,
+                    fee: CONTRACT_EXEC_FEE * VSYS_PRECISION,
+                    feeScale: FEE_SCALE,
+                    timestamp: this.timeStamp,
+                    functionIndex: this.isSplit ? WITHDRAW_FUNCIDX_SPLIT : WITHDRAW_FUNCIDX,
+                    functionData: transaction.prepareWithdraw(this.contractId, this.address, BigNumber(this.amount).multipliedBy(this.tokenUnity)),
+                    signature: transaction.prepareExecContractSignature(this.contractId, this.isSplit ? WITHDRAW_FUNCIDX_SPLIT : WITHDRAW_FUNCIDX, transaction.prepareWithdraw(this.contractId, this.address, BigNumber(this.amount).multipliedBy(this.tokenUnity)), this.attachment, BigNumber(CONTRACT_EXEC_FEE * VSYS_PRECISION), this.feeScale, BigNumber(this.timeStamp), this.getKeypair(this.addresses[this.address]).privateKey)
+                }
+                apiSchema = dataInfo
+            } else if (walletType === 'coldWallet') {
+                const coldDataInfo = {
+                    contractId: this.contractId,
+                    // senderPublicKey: this.coldAddresses[this.address].publicKey,
+                    fee: CONTRACT_EXEC_FEE * VSYS_PRECISION,
+                    feeScale: FEE_SCALE,
+                    timestamp: this.dataObject.timestamp,
+                    functionIndex: this.isSplit ? WITHDRAW_FUNCIDX_SPLIT : WITHDRAW_FUNCIDX,
```

Comment:
for all, if these variables all need to rewrite the formula, what is the usage of dataObject.

it is confused that some value use the value in dataObject, but...

Reply:
DataObject here is used for creating coldSignature, it'll be passed to coldWallet.
The reason using the value in dataObject is that the data in coldSignature should be the same as the data in apiSchema, otherwise, this transaction will fail.

Reply:
Sure. I know the usage here. But fee, feeScale, and all parameters should be same, but only part of them use the value from the dataObject

Reply:
daata have been used from the dataObject